### PR TITLE
[WIP] implement per key sync preference roughly

### PR DESCRIPTION
**Pull request by daquexian**
For issue #1847 ，implement per key sync preference roughly
## Description

- add a column _sync_ in table _updated_keys_.

- When creating a new key, the value of checkbox _"Synchronize with the Internet"_ will be stored in database. 

- In turn, when editing identities of a key, this checkbox goes disabled, following the value in database.

- Updating keys synchronized automatically when entering _ViewKeyActivity_

- Add a checkbox in _ViewKeyAdvShareFragment_ to modify the synchronize configuration. When clicking the checkbox to set a key synchronized, update and upload will be triggered.

I just implement this feature roughly, there is still something I need do. I open this PR to ask for some advice :) Is there some defect in this auto sync mechanism I designed?
 
## Motivation and Context

#1847 

## How Has This Been Tested?

I tested it for some keys, and ran unit test.

## Types of changes

- ✅ Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by daquexian** [comment]
@dschuermann what do you think of this? It's a large change so I'm looking forward to some guidance.
Is it better to update keys every time entering ViewKeyActivity or with a fixed frequency(e.g. every 3 days)?

**Comment by dschuermann** [comment]
@daquexian sorry, haven't freed up time to look into this. Will try to review it later today.

**Comment by daquexian** [comment]
@dschuermann It's alright :)  I'm not in a rush. I can work on another issue first.

File: OpenKeychain/src/main/java/org/sufficientlysecure/keychain/provider/KeychainDatabase.java [hunk 1]
```diff
@@ -325,6 +326,13 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
                 // splitUserId changed: Execute consolidate for new parsing of name, email
             case 18:
                 db.execSQL("ALTER TABLE keys ADD COLUMN is_secure INTEGER");
+            case 19:
+                Cursor res = db.rawQuery("Select * from updated_keys limit 1",null);
```
Comment (dschuermann) [inline OpenKeychain/src/main/java/org/sufficientlysecure/keychain/provider/KeychainDatabase.java]: Please uppercase SQL keywords for readability

File: OpenKeychain/src/main/java/org/sufficientlysecure/keychain/provider/KeychainDatabase.java [hunk 2]
```diff
@@ -325,6 +326,13 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
...
+                Log.d("aaa", "onUpgrade: " + res.getColumnIndex("sync"));
```
Comment (dschuermann) [inline OpenKeychain/src/main/java/org/sufficientlysecure/keychain/provider/KeychainDatabase.java]: remove

**Comment by Valodim** [comment]
closing stale PR

**Pull request closed without merging**